src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -409,16 +409,21 @@ ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
                                      UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
     if (uar->uar == NULL) {
 #if HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC
-        ucs_debug("UAR alloc WC failed. Fallback to NC\n");
+        ucs_debug("mlx5dv_devx_alloc_uar(device=%s, flags=0x%x(WC)) "
+                  "failed: %m, fallback to NC",
+                  uct_ib_device_name(&md->super.dev),
+                  UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
```

Comment:
maybe add function uct_ib_mlx5_devx_alloc_uar(md, flags, err_level=ERROR/DEBUG, char *title="WC"/"NC") which will wrap the debug/error print?